examples/decoding/plot_decoding_time_generalization.py

```diff
@@ -14,78 +14,80 @@
 """
 print(__doc__)
 
-# Authors: Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+# Authors: Jean-Remi King <jeanremi.king@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#          Denis Engemann <denis.engemann@gmail.com>
 #
 # License: BSD (3-clause)
-
 import numpy as np
-import matplotlib.pyplot as plt
 
 import mne
-from mne.datasets import spm_face
-from mne.decoding import time_generalization
-
-data_path = spm_face.data_path()
+from mne.datasets import sample
+from mne.decoding import GeneralizationAcrossTime
 
 ###############################################################################
+# Preprocess data
+
+data_path = sample.data_path()
 # Load and filter data, set up epochs
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+events_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
+raw = mne.io.Raw(raw_fname, preload=True)
+picks = mne.pick_types(raw.info, meg=True, exclude='bads')  # Pick MEG channels
+raw.filter(2, 30, method='fft', l_trans_bandwidth=0.1)  # Band pass filtering signals
+events = mne.read_events(events_fname)
+event_id = {'AudL': 1, 'AudR': 2, 'VisL': 3, 'VisR': 4}
+decim = 3  # decimate to make the example faster to run
+epochs = mne.Epochs(raw, events, event_id, -0.050, 0.400, proj=True,
+                    picks=picks, baseline=None, preload=True,
+                    reject=dict(mag=5e-12), decim=decim, verbose=False)
+epochs.equalize_event_counts(event_id, copy=False)
 
-raw_fname = data_path + '/MEG/spm/SPM_CTF_MEG_example_faces%d_3D_raw.fif'
+###############################################################################
+# Generalization across time (GAT)
 
-raw = mne.io.Raw(raw_fname % 1, preload=True)  # Take first run
-raw.append(mne.io.Raw(raw_fname % 2, preload=True))  # Take second run too
+# The function implements the method used in:
+# King, Gramfort, Schurger, Naccache & Dehaene, "Two distinct dynamic modes
+# subtend the detection of unexpected sounds", PLOS ONE, 2013
 
-picks = mne.pick_types(raw.info, meg=True, exclude='bads')
-raw.filter(1, 45, method='iir')
+# Define events of interest
+events = epochs.events[:, 2]
+viz_vs_auditory = np.in1d(events, (1, 2)).astype(int)
 
-events = mne.find_events(raw, stim_channel='UPPT001')
-event_id = {"faces": 1, "scrambled": 2}
-tmin, tmax = -0.1, 0.5
+gat = GeneralizationAcrossTime(predict_mode='cross-validation')
 
-# Set up pick list
-picks = mne.pick_types(raw.info, meg=True, eeg=False, stim=True, eog=True,
-                       ref_meg=False, exclude='bads')
+# fit and score
+gat.fit(epochs, y=viz_vs_auditory)
+gat.score(epochs, y=viz_vs_auditory)
+gat.plot_diagonal()  # plot decoding across time (correspond to GAT diagonal)
+gat.plot()  # plot full GAT matrix
```

Comment:
I get this:

`<GAT | fitted: from -0.053279 s. to 0.399590 s. (0.012939 s. every 0.012939 s.), predicted: 35 trials (predict) on 35 time slice(s) each, scored: True (accuracy_score)>`

some info are redundant also 3 digits after commas is enough. How about :

`<GAT | fit: from -0.053 s. to 0.400 s. (width 0.013 s. step 0.013 s.), predict: 35 trials on 35 time slice(s), scored: True (accuracy_score)>`